redbot/cogs/audio/audio.py

```diff
@@ -2212,6 +2212,7 @@ def _restart_connect(self):
                 player.add(ctx.author, search_choice)
             else:
                 return await self._embed_msg(ctx, _("Track exceeds maximum length."))
+        player.add(ctx.author, search_choice)
```

Comment:
Shouldn't this be garded behind an `else:`? Otherwise when maxlength is set and the track is under the max length, you'll be adding the song twice